### PR TITLE
Add supervision endpoints to reset directory elements index in ES

### PR DESCRIPTION
**Pull request by sBouzols**

### CODE REVIEW

File: src/test/java/org/gridsuite/directory/server/DirectoryTest.java [hunk 1]
```diff
@@ -1182,6 +1182,93 @@ public void testElementUpdateNotification() throws Exception {
         assertEquals(userMakingModification, updatedElement.getLastModifiedBy());
     }
 
+    @Test
+    public void testSupervision() throws Exception {
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/directory/server/DirectoryTest.java:1186]: Move it the `SupervisionTest` class ?

File: src/main/java/org/gridsuite/directory/server/services/DirectoryElementInfosService.java [hunk 2]
```diff
@@ -18,6 +19,7 @@
 import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.UUID;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/services/DirectoryElementInfosService.java:22]: Remove fields unused

File: src/main/java/org/gridsuite/directory/server/services/DirectoryElementInfosService.java [hunk 3]
```diff
@@ -36,8 +38,11 @@ public class DirectoryElementInfosService {
     static final String ELEMENT_OWNER = "owner.keyword";
     static final String ELEMENT_PRIVATE_STATUS = "isPrivate";
 
-    public DirectoryElementInfosService(ElasticsearchOperations elasticsearchOperations) {
+    private final DirectoryElementInfosRepository directoryElementInfosRepository;
```
[the remark this answers is not in the document]
Comment (sBouzols) [reply src/main/java/org/gridsuite/directory/server/services/DirectoryElementInfosService.java]: done

File: src/main/java/org/gridsuite/directory/server/services/DirectoryRepositoryService.java [hunk 4]
```diff
@@ -90,10 +95,20 @@ public boolean canRead(UUID id, String userId) {
         return directoryElementRepository.existsByIdAndOwnerOrIsPrivateAndId(id, userId, false, id);
     }
 
-    public void reindexAllElements() {
-        saveElementsInfos(directoryElementRepository.findAllByStashed(false).stream()
+    public void reindexElements(UUID directoryUuid) {
+        Optional<DirectoryElementEntity> dir = directoryElementRepository.findByIdAndType(directoryUuid, DIRECTORY);
```
[the remark this answers is not in the document]
Comment (sBouzols) [reply src/main/java/org/gridsuite/directory/server/services/DirectoryRepositoryService.java]: nor useful anymore

**Review by SlimaneAmar — APPROVED** [review]

**Review by antoinebhs — APPROVED** [review]

**Pull request merged**